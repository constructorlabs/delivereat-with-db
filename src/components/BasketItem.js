import React from 'react';
import PropTypes from 'prop-types';
import '../styles/BasketItem.scss';

function BasketItem({ menu, addToOrder, removeFromOrder, basketItem }) {
  const { name } = menu.find((item) => item.id === basketItem[0]);
  const { price } = menu.find((item) => item.id === basketItem[0]);

  return (
    <div className="basket-item">
      <div className="basket-item__title">
        <div className="basket-item__name">{name}</div>
        <div className="basket-item__buttons">
          <button
            className="basket-item__remove-button"
            type="button"
            onClick={() => removeFromOrder(basketItem[0])}
          >
            <i className="fas fa-minus-circle" />
          </button>
          <div className="basket-item__count">{basketItem[1]}</div>
          <button
            className="basket-item__add-button"
            type="button"
            onClick={() => addToOrder(basketItem[0])}
          >
            <i className="fas fa-plus-circle" />
          </button>
          <div className="basket-item__line-total">{`Â£${(price * basketItem[1]).toFixed(2)}`}</div>
        </div>
      </div>
    </div>
  );
}

BasketItem.propTypes = {
  menu: PropTypes.array.isRequired,
  basketItem: PropTypes.array.isRequired,
  addToOrder: PropTypes.func.isRequired,
  removeFromOrder: PropTypes.func.isRequired,
};

export default BasketItem;
