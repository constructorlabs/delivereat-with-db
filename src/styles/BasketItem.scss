@import "parameters";

.item {
  display: grid;
  grid-template-columns: 1fr 4fr 1fr;
  justify-content: center;
  align-items: center;

  %block {
    display: flex;
    justify-self: center;
    align-self: center;
  }
  p {
    @extend %block;
  }

  .quantity {
    @extend %block;
    flex-direction: row;
  }
}
