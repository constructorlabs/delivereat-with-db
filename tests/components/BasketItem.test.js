import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BasketItem from '../../src/components/BasketItem';

const basketItem = [1, 2];
const menu = [
  {
    id: 1,
    name: 'Espresso',
    description: 'Single shot, 30ml',
    price: '2.00',
  },
  {
    id: 2,
    name: 'Ristretto',
    description: 'Concentrated espresso, 22ml',
    price: '2.00',
  },
  {
    id: 3,
    name: 'Macchiato',
    description: '60ml espresso + a dab of milk foam',
    price: '2.20',
  },
];

const mockAddToOrder = jest.fn();
const mockRemoveFromOrder = jest.fn();

describe('Basket Item', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <BasketItem
          menu={menu}
          basketItem={basketItem}
          addToOrder={mockAddToOrder}
          removeFromOrder={mockRemoveFromOrder}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('click removes item', () => {
    const wrapper = shallow(
      <BasketItem
        menu={menu}
        basketItem={basketItem}
        addToOrder={mockAddToOrder}
        removeFromOrder={mockRemoveFromOrder}
      />,
    );
    wrapper.find('.basket-item__remove-button').simulate('click');
    expect(mockRemoveFromOrder.mock.calls).toEqual([[1]]);
  });
  test('click adds item', () => {
    const wrapper = shallow(
      <BasketItem
        menu={menu}
        basketItem={basketItem}
        addToOrder={mockAddToOrder}
        removeFromOrder={mockRemoveFromOrder}
      />,
    );
    wrapper.find('.basket-item__add-button').simulate('click');
    expect(mockAddToOrder.mock.calls).toEqual([[1]]);
  });
});
