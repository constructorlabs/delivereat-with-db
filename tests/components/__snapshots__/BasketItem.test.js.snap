// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket Item matches the snapshot 1`] = `
<div
  className="basket-item"
>
  <div
    className="basket-item__title"
  >
    <div
      className="basket-item__name"
    >
      Espresso
    </div>
    <div
      className="basket-item__buttons"
    >
      <button
        className="basket-item__remove-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-minus-circle"
        />
      </button>
      <div
        className="basket-item__count"
      >
        2
      </div>
      <button
        className="basket-item__add-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-plus-circle"
        />
      </button>
      <div
        className="basket-item__line-total"
      >
        Â£4.00
      </div>
    </div>
  </div>
</div>
`;
